mail_tracking_mailgun/readme/CONFIGURE.rst

```diff
@@ -1,27 +1,11 @@
-You must configure Mailgun webhooks in order to receive mail events:
-
-1. Got a Mailgun account and validate your sending domain.
-2. Go to Webhook tab and configure the below URL for each event:
-
-.. code:: html
-
-   https://<your_domain>/mail/tracking/all/<your_database>
-
-Replace '<your_domain>' with your Odoo install domain name
-and '<your_database>' with your database name.
-
-In order to validate Mailgun webhooks you have to configure the following system
-parameters:
-
-- `mailgun.apikey`: You can find Mailgun api_key in your validated sending
-  domain.
-- `mailgun.api_url`: It should be fine as it is, but it could change in the
-  future.
-- `mailgun.domain`: In case your sending domain is different from the one
-  configured in `mail.catchall.domain`.
-- `mailgun.validation_key`: If you want to be able to check mail address
-  validity you must config this parameter with your account Public Validation
-  Key.
+To configure this module, you need to:
+
+#. Go to Mailgun, create an account and validate your sending domain.
+#. Go back to Odoo.
+#. Go to *Settings > General Settings > Discuss > Enable mail tracking with Mailgun*.
+#. Fill all the values. The only one required is the API key.
+#. Optionally click *Unregister Mailgun webhooks* and accept.
```

Comment:
You shouldn't be required to press these buttons (they should be only for manual actions), but on change of the parameter, you should first before changing data to unregister hooks, and then change them and register hooks.

Reply:
> 1.  I don't want to override the setting/getting of ir.config_parameter, as it is very widely used. And that'd be the only way to get this properly automated.

That's because you are thinking of the field name automatism on upstream, but using another field name, you can do the set and get you want.

> 2. This could unregister webhooks that are not from Odoo.

So you are saying that clicking manually on this unregister other webhooks not created by Odoo??! That's the problem, and you should know (by the name or something), which ones were created by Odoo and only unregister that ones.

> 3. Infrastructure configuration should be predictable.

Yes, that's why you should automate it. If not, Ansible wouldn't exist.

> 4. Doing this manually implicitly helps checking that your api key is valid.

And if not valid, you receive an error when saving, so that's the desired effect of not storing something invalid.

So I still think you should do it automatically.